pilot/cmd/pilot-agent/main.go

```diff
@@ -519,15 +519,16 @@ var (
 // dedupes the string array and also ignores the empty string.
 func dedupeStrings(in []string) []string {
 	stringMap := map[string]bool{}
+	unique := make([]string, 0)
 	for _, c := range in {
 		if len(c) > 0 {
-			stringMap[c] = true
+			if _, ok := stringMap[c]; !ok {
```

Comment:
if !stringMap[c]

Reply:
accept

Reply:
"if !stringMap[c]" and "if _, ok := stringMap[c]; !ok " are both ok, why change to the first?